src/check_run.c

```diff
@@ -158,8 +159,47 @@ static void srunner_run_end(SRunner * sr,
     set_fork_status(CK_FORK);
 }
 
+/*
+ * Helper func to compare two lists of tags and return true if there is a
+ * common tag.
+ */
+static unsigned int matching_tag(const char *tags1_orig, const char *tags2_orig)
+{
+
+    char *tags1;
+    char *tag1;
+
+    if ((tags1_orig == NULL) || (tags2_orig == NULL))
+	return 0;
+
+    tags1 = strdup(tags1_orig);
+    tag1 = strtok(tags1, " ");
```

Comment:
Can strtok() be used in the way it is being used here? Namely, strtok() maintains state on its original string. However, strtok is being used here to maintain state on two different strings, namely tags1_orig and tags2_orig. Should strtok_r be used instead to avoid issues?